Makefile

```diff
@@ -47,40 +42,64 @@ devel: test
 
 .PHONY: package
 package: test
-	cabal check
-	# tests that a source-distribution can be generated
-	cabal sdist
+	# check and bundle the package
+	stack sdist
+	# outputs to .stack-work/install/<machine-architecture>/<snapshot>/doc
+
 	# check that the generated source-distribution can be built & installed
+	# TODO: where to put build?
 	{ \
-	set -e; \
-	export SRC_TGZ=$$(cabal info . | awk '{print $$2 ".tar.gz";exit}') ; \
-	cd dist/; \
-	cabal sandbox init; \
-	if [ -f "$$SRC_TGZ" ]; then \
-		cabal install --force-reinstalls "$$SRC_TGZ"; \
-	else \
-		echo "expected '$$SRC_TGZ' not found"; \
-		exit 1; \
-	fi; \
+	set -e; set -x; \
+	export SRC_TGZ=$$(stack sdist 2>&1 | tail -n 1) ; \
+	export PACKAGE=$${SRC_TGZ%.tar.gz}; \
+	export PACKAGE=$${PACKAGE##*/}; \
+	mkdir -p .build/; cd .build/; \
+	cp $$SRC_TGZ .; \
+	tar xf $$SRC_TGZ; \
+	cd $$PACKAGE; \
+	pwd; \
+	stack init; \
+	stack build; \
+	stack test; \
+	cd ../..; \
+	rm -r .build; \
```

Comment:
@power-fungus i imagine the `tar` is within `.build`? if so, this `rm` should be in `clean`.
that would allow us to upload package after building it

Reply:
This was a proof of concept. But keeping the build-directory seems reasonable. Additionally, I have no idea where to put the `tar`. Is the `.build` directory ok? Or should it be called something like "release", where everything is put, including documentation?

Reply:
@power-fungus cabal by default puts it under `dist/`, we could reuse that.

Reply:
and given `dist/` is already in `.gitignore` we don't need to delete it either